src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,120 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { PeopleTable } from '../components/PeopleTable';
+import { Loader } from '../components/Loader';
+import { getPeople } from '../api';
+import { PersonType } from '../types';
+import { Filter } from '../types/Filter';
+import { ORDER_PARAM, SORT_PARAM, Sort } from '../types/Sort';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<PersonType[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [searchParams] = useSearchParams();
+
+  function loadPeople() {
+    getPeople()
+      .then(setPeople)
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }
+
+  useEffect(loadPeople, []);
+
+  function getFilteredPeople(): PersonType[] | null {
+    let filteredPeople = [...people];
+    const sort = searchParams.get(SORT_PARAM);
+    const desc = searchParams.get(ORDER_PARAM);
+    const sex = searchParams.get(Filter.SEX);
+    const name = searchParams.get(Filter.NAME);
+    const centuries = searchParams.getAll(Filter.CENTURIES);
+
+    if (sort) {
+      switch (sort) {
+        case Sort.SEX:
+        case Sort.NAME:
+          filteredPeople.sort((a, b) => (
+            a[sort].localeCompare(b[sort])));
+          break;
+
+        case Sort.BORN:
+        case Sort.DIED:
+          filteredPeople.sort((a, b) => (a[sort] - b[sort]));
+          break;
+
+        default:
+          break;
+      }
+
+      if (desc) {
+        return filteredPeople.reverse();
+      }
+    }
+
+    if (sex) {
+      filteredPeople = filteredPeople.filter(person => person.sex === sex);
+    }
+
+    if (name) {
+      filteredPeople = filteredPeople.filter(person => person.name.toLowerCase()
+        .includes(name.toLowerCase())
+        || person.fatherName?.toLowerCase().includes(name.toLowerCase())
+        || person.motherName?.includes(name.toLowerCase()));
+    }
+
+    if (centuries) {
```

Comment:
you iterate with array so many times. 
You filter filtered values that were filtered before. It's not good. 
You should filter an array as less as possible. Try to make it only once. And in the process you filter you can add conditions inside how to filter